presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -774,20 +775,15 @@ public HashComputationSet withHashComputation(Optional<HashComputation> hashComp
 
     private static class HashComputation
     {
-        private final List<Symbol> fields;
+        private final Set<Symbol> fields;
 
         private HashComputation(Iterable<Symbol> fields)
         {
             requireNonNull(fields, "fields is null");
-            this.fields = ImmutableList.copyOf(fields);
+            this.fields = ImmutableSortedSet.copyOf(fields);
```

Comment:
This assumes that the names given to fields across all places in the plan that might need to compute the hash (e.g., two branches of a join) are consistent. Let me see if I can come up with an example query.